nlprule/src/components/tokenizer/mod.rs

```diff
@@ -0,0 +1,238 @@
+//! A tokenizer to split raw text into tokens.
+//! Tokens are assigned lemmas and part-of-speech tags by lookup from a [Tagger][tag::Tagger] and chunks containing
+//! information about noun / verb and grammatical case by a statistical [Chunker][chunk::Chunker].
+//! Tokens are *disambiguated* (i. e. information from the initial assignment is changed) in a rule-based way by
+//! [DisambiguationRule][crate::rule::DisambiguationRule]s.
+
+#[cfg(feature = "compile")]
+mod compile;
+
+use std::ops::Range;
+
+use crate::types::*;
+use crate::{properties::*, utils::regex::Regex};
+use lazy_static::lazy_static;
+use once_cell::sync::OnceCell;
+use serde::{Deserialize, Serialize};
+
+use super::{tagger::Tagger, Component};
+
+/// Split a text at the points where the given function is true.
+/// Keeps the separators. See https://stackoverflow.com/a/40296745.
+fn split<F>(text: &str, split_func: F) -> Vec<&str>
```

Comment:
Is the assumption of a single character always true? There are the unicode bold variants of `?!.`